Sources/MapboxCoreNavigation/VisualInstructionComponentSubType.swift

```diff
@@ -0,0 +1,32 @@
+import MapboxDirections
+import MapboxNavigationNative
+
+extension VisualInstruction.Component.SubType {
+    
+    init(_ native: MapboxNavigationNative.BannerComponentSubType) {
+        switch native {
+        case .JCT:
+            self = .jct
+        case .signboard:
+            self = .signboard
+        case .sapaguidemap:
+            self = .sapaGuideMap
+        case .sapa:
+            self = .sapa
+        case .aftertoll:
+            self = .afterToll
+        case .cityreal:
+            self = .cityReal
+        case .ent:
+            self = .expresswayEntrance
+        case .exit:
+            self = .expresswayExit
+        case .tollbranch:
+            self = .tollBranch
+        case .directionboard:
+            self = .directionBoard
+        @unknown default:
+            fatalError("Unknown BannerComponentSubType value.")
```

Comment:
Similar to https://github.com/mapbox/mapbox-navigation-ios/pull/4300#discussion_r1047995676: should we be failing here? There is no `undefined` type to default to though.

Reply:
We handle such cases in our codebase by either using `fatalError` or `preconditionFailure`. Since there is no fallback to default we'll probably have to use either of them (I'm personally open to any).

Reply:
This makes sense. I would stay with `fatalError` then so that it always prints an error message.